File: java/parsers/src/test/java/jp/co/teruuu/parser/json/JNullParserTest.java
package jp.co.teruuu.parser.json;

import jp.co.teruuu.parser.common.ParseResult;
import jp.co.teruuu.parser.json.type.JBoolean;
import jp.co.teruuu.parser.json.type.JNull;
import org.junit.Test;

public class JNullParserTest {

    @Test
    public void test() {
        JParser parser = new JNullParser();

        ParseResult result1 = parser.parse("null");
        assert(result1 instanceof ParseResult.Success);
        assert((ParseResult.Success) result1).value.equals(new JNull());

        ParseResult result2 = parser.parse("NULL");
        assert(result2 instanceof ParseResult.Failure);
    }
}

File: scala/parsers/src/test/scala/jp/co/teruuu/parser/json/JNullParserTest.scala
package jp.co.teruuu.parser.json

import jp.co.teruuu.parser.common.{ParseNg, ParseOk}
import jp.co.teruuu.parser.json.`type`.{JBool, JNull}
import org.scalatest.FlatSpec

class JNullParserTest extends FlatSpec{

  "parse" should "successt" in {
    val parser = JParser.nu()

    val result1 = parser.parse("null")
    assert(result1.isInstanceOf[ParseOk[JNull]])
    assert(result1.asInstanceOf[ParseOk[JNull]].value.equals(JNull()))
  }

  "parse" should "fail" in {
    val parser = JParser.nu()

    val result1 = parser.parse("NULL")
    assert(result1.isInstanceOf[ParseNg[JNull]])
  }
}
